myuw/dao/enrollment.py

```diff
@@ -48,6 +71,77 @@ def get_main_campus(request):
     return campuses
 
 
+def _get_degrees_for_terms(terms, enrollments, accessor):
+    """
+    Takes in a list of terms and a dictionary of terms to enrollments (returned
+    by get_all_enrollments), and returns a list of either majors or minors
+    and their terms, depending upon which accessor is used, 'majors' or
+    'minors'
+    """
+    degrees = []
+    previous = None
+
+    for term in terms:
+        if (term in enrollments and
+                len(getattr(enrollments[term], accessor)) > 0):
+            enrollment = enrollments[term]
+            entry = {}
+            entry['quarter'] = term.quarter
+            entry['year'] = term.year
+
+            term_degrees = getattr(enrollments[term], accessor)
+            entry[accessor] = []
+
+            if previous is not None:
+                entry['same_as_previous'] = _compare_degrees(previous,
+                                                             term_degrees)
+            else:
+                entry['same_as_previous'] = True
+
+            for degree in term_degrees:
+                entry[accessor].append(degree.json_data())
+
+            degrees.append(entry)
+            previous = term_degrees
+
+    return degrees
+
+
+def get_majors_for_terms(terms, enrollments):
+    """
+    Takes in a list of terms and a dictionary of terms to enrollments (returned
+    by get_all_enrollments), and returns a list of majors and their terms
+    """
+    return _get_degrees_for_terms(terms, enrollments, 'majors')
+
+
+def get_minors_for_terms(terms, enrollments):
+    """
+    Takes in a list of terms and a dictionary of terms to enrollments (returned
+    by get_all_enrollments), and returns a list of minors and their terms
+    """
+    return _get_degrees_for_terms(terms, enrollments, 'minors')
+
+
+def _compare_degrees(first, second):
+    """
+    Takes in two lists of degrees (either major or minors) and checks to see
+    if they are the same. Returns True if so, False if they are different
+    """
+    if len(first) != len(second):
+        return False
+
+    for entry in first:
```

Comment:
Use a dictionary here, so you can iterate over list1, then list2, vs doing list1 * list2.